tests/api/sip/test_client.py

```diff
@@ -1,9 +1,12 @@
 """Standalone tests of the SIP2 client."""
-import os
 import socket
 import ssl
+import tempfile
+from typing import List, Optional
+from unittest.mock import MagicMock, Mock
 
 import pytest
+from _pytest.monkeypatch import MonkeyPatch
```

Comment:
We're relying on a member of a non-public package, which could break without warning. This is discouraged and I think we should do it only where there is a compelling case for it (and I think we should clearly document these cases when they occur). Do we really need it here?

Reply:
I think I disagree on this, for the same reasons as above. This is just used as an annotation, and I think its preferable to have an accurate type annotation.

Reply:
In cases like this, it seems like we should be able to skip the annotation and have the types propagate through. In many cases, we should just get the type without even needing to supply the annotation.